File: data/scripts/ini3.py
filename = "dataset.txt"
current = 0
stringa = ""
a, b, c, d = 0, 0, 0, 0
with open(filename) as f:
    for riga in f:
        if current == 0:
           stringa = riga
           current += 1
        elif current == 1:
           lista = riga.split(" ")
           a, b, c, d = int(lista[0]), int(lista[1]), int(lista[2]), int(lista[3])
           current += 1
        elif current == 2:
           break   
#print(stringa)
#print(a, b, c, d)
output = f"{stringa[a:b+1]} {stringa[c:d+1]}"
print(output)

File: public/js-solutions/ini3.mjs
// ini3 - esercizio introduttivo "Python Village": data una stringa e
// quattro interi a b c d, restituisce le due sottostringhe [a, b] e
// [c, d] (indici inclusivi, 0-based) separate da uno spazio. Soluzione
// JavaScript indipendente, non una trascrizione di problem.py, riscritta
// in modo idiomatico per JS.
//
// Contratto: riceve il contenuto testuale di dataset.txt, restituisce
// l'output testuale (stessa forma dell'output Python).
export default function solve(datasetText) {
  const righe = datasetText.split("\n");
  const stringa = righe[0]?.replace(/\r$/, "") ?? "";
  const parti = (righe[1] ?? "").trim().split(/\s+/);

  if (!stringa || parti.length !== 4 || parti.some((p) => !/^\d+$/.test(p))) {
    throw new Error("Input non valido: attese una stringa e quattro interi a b c d");
  }

  const [a, b, c, d] = parti.map(Number);
  return `${stringa.slice(a, b + 1)} ${stringa.slice(c, d + 1)}\n`;
}
